src/order/bounded_lattice.lean

```diff
@@ -271,39 +297,53 @@ end semilattice_inf_bot
 /-- A bounded lattice is a lattice with a top and bottom element,
   denoted `⊤` and `⊥` respectively. This allows for the interpretation
   of all finite suprema and infima, taking `inf ∅ = ⊤` and `sup ∅ = ⊥`. -/
-class bounded_lattice (α : Type u) extends lattice α, order_top α, order_bot α
+class bounded_lattice (α : Type u) extends lattice α, has_top α, has_bot α :=
+(le_top : ∀ a : α, a ≤ ⊤)
+(bot_le : ∀ a : α, ⊥ ≤ a)
 
 @[priority 100] -- see Note [lower instance priority]
 instance semilattice_inf_top_of_bounded_lattice (α : Type u) [bl : bounded_lattice α] :
   semilattice_inf_top α :=
-{ le_top := λ x, @le_top α _ x, ..bl }
+{ ..bl }
 
 @[priority 100] -- see Note [lower instance priority]
 instance semilattice_inf_bot_of_bounded_lattice (α : Type u) [bl : bounded_lattice α] :
   semilattice_inf_bot α :=
-{ bot_le := λ x, @bot_le α _ x, ..bl }
+{ ..bl }
 
 @[priority 100] -- see Note [lower instance priority]
 instance semilattice_sup_top_of_bounded_lattice (α : Type u) [bl : bounded_lattice α] :
   semilattice_sup_top α :=
-{ le_top := λ x, @le_top α _ x, ..bl }
+{ ..bl }
 
 @[priority 100] -- see Note [lower instance priority]
 instance semilattice_sup_bot_of_bounded_lattice (α : Type u) [bl : bounded_lattice α] :
   semilattice_sup_bot α :=
-{ bot_le := λ x, @bot_le α _ x, ..bl }
+{ ..bl }
 
+-- tougher, because the has_le required for order_top/bot is now different
```

Comment:
```suggestion
```
This is a comment related to this change, but won't make sense after this PR gets accepted, right?